crates/cargo-contract/src/cmd/verify.rs

```diff
@@ -0,0 +1,236 @@
+
+            // Note, assuming both versions of `cargo-contract` were installed with the
+            // same lockfile (e.g `--locked`) then the versions of `wasm-opt`
+            // should also match.
+            let cargo_contract_matches =
+                cargo_contract_version == expected_cargo_contract_version;
+            let mismatched_cargo_contract = format!(
+                "\nYou are trying to `verify` a contract using `cargo-contract` version \
+            `{cargo_contract_version}`.\n\
+             However, the original contract was built using `cargo-contract` version \
+             `{expected_cargo_contract_version}`.\n\
+             Please install the matching version and re-run the `verify` command.",
+            );
+            anyhow::ensure!(
+                cargo_contract_matches,
+                mismatched_cargo_contract.bright_yellow()
+            );
+        }
+
+        // 3a. Call `cargo contract build` with the `BuildInfo` from the metadata.
+        let args = ExecuteArgs {
+            manifest_path: manifest_path.clone(),
+            verbosity,
+            build_mode,
+            build_artifact: BuildArtifacts::All,
+            optimization_passes: Some(build_info.wasm_opt_settings.optimization_passes),
+            keep_debug_symbols: build_info.wasm_opt_settings.keep_debug_symbols,
+            image: ImageVariant::from(metadata.image.clone()),
+            dylint: false,
+            ..Default::default()
+        };
+
+        let build_result = execute(args)?;
+
+        // 4. Grab the code hash from the built contract and compare it with the reference
+        //    one.
+        let reference_code_hash = metadata.source.hash;
+        let built_contract_path = if let Some(m) = build_result.metadata_result {
+            m
+        } else {
+            // Since we're building the contract ourselves this should always be
+            // populated, but we'll bail out here just in case.
+            anyhow::bail!(
+                "\nThe metadata for the workspace contract does not contain a Wasm binary,\n\
+                therefore we are unable to verify the contract."
+                .to_string()
+                .bright_yellow()
+            )
+        };
+
+        let target_bundle = built_contract_path.dest_bundle;
+
+        let file = File::open(target_bundle.clone()).context(format!(
+            "Failed to open contract bundle {}",
+            target_bundle.display()
+        ))?;
+        let built_contract: ContractMetadata =
+            serde_json::from_reader(file).context(format!(
+                "Failed to deserialize contract bundle {}",
+                target_bundle.display()
+            ))?;
+
+        let target_code_hash = built_contract.source.hash;
+
+        if reference_code_hash != target_code_hash {
+            tracing::debug!(
+                "Expected Code Hash: '{:?}'\n\nGot Code Hash: `{:?}`",
```

Comment:
I'd like to see the actual code hashes here displayed on the console, instead of hidden in the logs.